// @letta-ai/letta-client/serialization/resources/identities/client/requests/IdentityUpsert.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.IdentityUpsert = void 0;
const core = __importStar(require("../../../../../core"));
const IdentityType_1 = require("../../../../types/IdentityType");
const IdentityProperty_1 = require("../../../../types/IdentityProperty");
exports.IdentityUpsert = core.serialization.object({
    identifierKey: core.serialization.property("identifier_key", core.serialization.string()),
    name: core.serialization.string(),
    identityType: core.serialization.property("identity_type", IdentityType_1.IdentityType),
    projectId: core.serialization.property("project_id", core.serialization.string().optional()),
    agentIds: core.serialization.property("agent_ids", core.serialization.list(core.serialization.string()).optional()),
    blockIds: core.serialization.property("block_ids", core.serialization.list(core.serialization.string()).optional()),
    properties: core.serialization.list(IdentityProperty_1.IdentityProperty).optional(),
});
