// @letta-ai/letta-client/serialization/resources/templates/types/TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItem.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItem = void 0;
const core = __importStar(require("../../../../core"));
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemDescription_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemDescription");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemInstructions_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemInstructions");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemMetadata_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemMetadata");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemId_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemId");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfig_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfig");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemOrganizationId_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemOrganizationId");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemVectorDbProvider_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemVectorDbProvider");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemCreatedById_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemCreatedById");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemLastUpdatedById_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemLastUpdatedById");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemCreatedAt_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemCreatedAt");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemUpdatedAt_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemUpdatedAt");
exports.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItem = core.serialization.object({
    name: core.serialization.string(),
    description: TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemDescription_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemDescription.optional(),
    instructions: TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemInstructions_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemInstructions.optional(),
    metadata: TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemMetadata_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemMetadata.optional(),
    id: TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemId_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemId.optional(),
    embeddingConfig: core.serialization.property("embedding_config", TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfig_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfig),
    organizationId: core.serialization.property("organization_id", TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemOrganizationId_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemOrganizationId.optional()),
    vectorDbProvider: core.serialization.property("vector_db_provider", TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemVectorDbProvider_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemVectorDbProvider.optional()),
    createdById: core.serialization.property("created_by_id", TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemCreatedById_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemCreatedById.optional()),
    lastUpdatedById: core.serialization.property("last_updated_by_id", TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemLastUpdatedById_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemLastUpdatedById.optional()),
    createdAt: core.serialization.property("created_at", TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemCreatedAt_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemCreatedAt.optional()),
    updatedAt: core.serialization.property("updated_at", TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemUpdatedAt_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemUpdatedAt.optional()),
});
