// @letta-ai/letta-client/serialization/resources/templates/types/TemplatesCreateAgentsFromTemplateResponseGroup.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplatesCreateAgentsFromTemplateResponseGroup = void 0;
const core = __importStar(require("../../../../core"));
const TemplatesCreateAgentsFromTemplateResponseGroupManagerType_1 = require("./TemplatesCreateAgentsFromTemplateResponseGroupManagerType");
const TemplatesCreateAgentsFromTemplateResponseGroupProjectId_1 = require("./TemplatesCreateAgentsFromTemplateResponseGroupProjectId");
const TemplatesCreateAgentsFromTemplateResponseGroupTemplateId_1 = require("./TemplatesCreateAgentsFromTemplateResponseGroupTemplateId");
const TemplatesCreateAgentsFromTemplateResponseGroupBaseTemplateId_1 = require("./TemplatesCreateAgentsFromTemplateResponseGroupBaseTemplateId");
const TemplatesCreateAgentsFromTemplateResponseGroupDeploymentId_1 = require("./TemplatesCreateAgentsFromTemplateResponseGroupDeploymentId");
const TemplatesCreateAgentsFromTemplateResponseGroupSharedBlockIds_1 = require("./TemplatesCreateAgentsFromTemplateResponseGroupSharedBlockIds");
const TemplatesCreateAgentsFromTemplateResponseGroupManagerAgentId_1 = require("./TemplatesCreateAgentsFromTemplateResponseGroupManagerAgentId");
const TemplatesCreateAgentsFromTemplateResponseGroupTerminationToken_1 = require("./TemplatesCreateAgentsFromTemplateResponseGroupTerminationToken");
const TemplatesCreateAgentsFromTemplateResponseGroupMaxTurns_1 = require("./TemplatesCreateAgentsFromTemplateResponseGroupMaxTurns");
const TemplatesCreateAgentsFromTemplateResponseGroupSleeptimeAgentFrequency_1 = require("./TemplatesCreateAgentsFromTemplateResponseGroupSleeptimeAgentFrequency");
const TemplatesCreateAgentsFromTemplateResponseGroupTurnsCounter_1 = require("./TemplatesCreateAgentsFromTemplateResponseGroupTurnsCounter");
const TemplatesCreateAgentsFromTemplateResponseGroupLastProcessedMessageId_1 = require("./TemplatesCreateAgentsFromTemplateResponseGroupLastProcessedMessageId");
const TemplatesCreateAgentsFromTemplateResponseGroupMaxMessageBufferLength_1 = require("./TemplatesCreateAgentsFromTemplateResponseGroupMaxMessageBufferLength");
const TemplatesCreateAgentsFromTemplateResponseGroupMinMessageBufferLength_1 = require("./TemplatesCreateAgentsFromTemplateResponseGroupMinMessageBufferLength");
const TemplatesCreateAgentsFromTemplateResponseGroupHidden_1 = require("./TemplatesCreateAgentsFromTemplateResponseGroupHidden");
exports.TemplatesCreateAgentsFromTemplateResponseGroup = core.serialization.object({
    id: core.serialization.string(),
    managerType: core.serialization.property("manager_type", TemplatesCreateAgentsFromTemplateResponseGroupManagerType_1.TemplatesCreateAgentsFromTemplateResponseGroupManagerType),
    agentIds: core.serialization.property("agent_ids", core.serialization.list(core.serialization.string())),
    description: core.serialization.string(),
    projectId: core.serialization.property("project_id", TemplatesCreateAgentsFromTemplateResponseGroupProjectId_1.TemplatesCreateAgentsFromTemplateResponseGroupProjectId.optional()),
    templateId: core.serialization.property("template_id", TemplatesCreateAgentsFromTemplateResponseGroupTemplateId_1.TemplatesCreateAgentsFromTemplateResponseGroupTemplateId.optional()),
    baseTemplateId: core.serialization.property("base_template_id", TemplatesCreateAgentsFromTemplateResponseGroupBaseTemplateId_1.TemplatesCreateAgentsFromTemplateResponseGroupBaseTemplateId.optional()),
    deploymentId: core.serialization.property("deployment_id", TemplatesCreateAgentsFromTemplateResponseGroupDeploymentId_1.TemplatesCreateAgentsFromTemplateResponseGroupDeploymentId.optional()),
    sharedBlockIds: core.serialization.property("shared_block_ids", TemplatesCreateAgentsFromTemplateResponseGroupSharedBlockIds_1.TemplatesCreateAgentsFromTemplateResponseGroupSharedBlockIds.optional()),
    managerAgentId: core.serialization.property("manager_agent_id", TemplatesCreateAgentsFromTemplateResponseGroupManagerAgentId_1.TemplatesCreateAgentsFromTemplateResponseGroupManagerAgentId.optional()),
    terminationToken: core.serialization.property("termination_token", TemplatesCreateAgentsFromTemplateResponseGroupTerminationToken_1.TemplatesCreateAgentsFromTemplateResponseGroupTerminationToken.optional()),
    maxTurns: core.serialization.property("max_turns", TemplatesCreateAgentsFromTemplateResponseGroupMaxTurns_1.TemplatesCreateAgentsFromTemplateResponseGroupMaxTurns.optional()),
    sleeptimeAgentFrequency: core.serialization.property("sleeptime_agent_frequency", TemplatesCreateAgentsFromTemplateResponseGroupSleeptimeAgentFrequency_1.TemplatesCreateAgentsFromTemplateResponseGroupSleeptimeAgentFrequency.optional()),
    turnsCounter: core.serialization.property("turns_counter", TemplatesCreateAgentsFromTemplateResponseGroupTurnsCounter_1.TemplatesCreateAgentsFromTemplateResponseGroupTurnsCounter.optional()),
    lastProcessedMessageId: core.serialization.property("last_processed_message_id", TemplatesCreateAgentsFromTemplateResponseGroupLastProcessedMessageId_1.TemplatesCreateAgentsFromTemplateResponseGroupLastProcessedMessageId.optional()),
    maxMessageBufferLength: core.serialization.property("max_message_buffer_length", TemplatesCreateAgentsFromTemplateResponseGroupMaxMessageBufferLength_1.TemplatesCreateAgentsFromTemplateResponseGroupMaxMessageBufferLength.optional()),
    minMessageBufferLength: core.serialization.property("min_message_buffer_length", TemplatesCreateAgentsFromTemplateResponseGroupMinMessageBufferLength_1.TemplatesCreateAgentsFromTemplateResponseGroupMinMessageBufferLength.optional()),
    hidden: TemplatesCreateAgentsFromTemplateResponseGroupHidden_1.TemplatesCreateAgentsFromTemplateResponseGroupHidden.optional(),
});
