// @letta-ai/letta-client/serialization/types/ChatCompletion.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChatCompletion = void 0;
const core = __importStar(require("../../core"));
const Choice_1 = require("./Choice");
const ChatCompletionServiceTier_1 = require("./ChatCompletionServiceTier");
const CompletionUsage_1 = require("./CompletionUsage");
exports.ChatCompletion = core.serialization
    .object({
    id: core.serialization.string(),
    choices: core.serialization.list(Choice_1.Choice),
    created: core.serialization.number(),
    model: core.serialization.string(),
    object: core.serialization.stringLiteral("chat.completion"),
    serviceTier: core.serialization.property("service_tier", ChatCompletionServiceTier_1.ChatCompletionServiceTier.optional()),
    systemFingerprint: core.serialization.property("system_fingerprint", core.serialization.string().optional()),
    usage: CompletionUsage_1.CompletionUsage.optional(),
})
    .passthrough();
