// @letta-ai/letta-client/serialization/types/ReasoningMessage.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReasoningMessage = void 0;
const core = __importStar(require("../../core"));
const ReasoningMessageSource_1 = require("./ReasoningMessageSource");
exports.ReasoningMessage = core.serialization.object({
    id: core.serialization.string(),
    date: core.serialization.date(),
    name: core.serialization.string().optional(),
    messageType: core.serialization.property("message_type", core.serialization.stringLiteral("reasoning_message")),
    otid: core.serialization.string().optional(),
    senderId: core.serialization.property("sender_id", core.serialization.string().optional()),
    stepId: core.serialization.property("step_id", core.serialization.string().optional()),
    isErr: core.serialization.property("is_err", core.serialization.boolean().optional()),
    seqId: core.serialization.property("seq_id", core.serialization.number().optional()),
    runId: core.serialization.property("run_id", core.serialization.string().optional()),
    source: ReasoningMessageSource_1.ReasoningMessageSource.optional(),
    reasoning: core.serialization.string(),
    signature: core.serialization.string().optional(),
});
