// @letta-ai/letta-client/serialization/types/Run.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Run = void 0;
const core = __importStar(require("../../core"));
const RunStatus_1 = require("./RunStatus");
const LettaRequestConfig_1 = require("./LettaRequestConfig");
const StopReasonType_1 = require("./StopReasonType");
exports.Run = core.serialization.object({
    id: core.serialization.string().optional(),
    status: RunStatus_1.RunStatus.optional(),
    createdAt: core.serialization.property("created_at", core.serialization.date().optional()),
    completedAt: core.serialization.property("completed_at", core.serialization.date().optional()),
    agentId: core.serialization.property("agent_id", core.serialization.string()),
    background: core.serialization.boolean().optional(),
    metadata: core.serialization.record(core.serialization.string(), core.serialization.unknown()).optional(),
    requestConfig: core.serialization.property("request_config", LettaRequestConfig_1.LettaRequestConfig.optional()),
    stopReason: core.serialization.property("stop_reason", StopReasonType_1.StopReasonType.optional()),
    callbackUrl: core.serialization.property("callback_url", core.serialization.string().optional()),
    callbackSentAt: core.serialization.property("callback_sent_at", core.serialization.date().optional()),
    callbackStatusCode: core.serialization.property("callback_status_code", core.serialization.number().optional()),
    callbackError: core.serialization.property("callback_error", core.serialization.string().optional()),
    ttftNs: core.serialization.property("ttft_ns", core.serialization.number().optional()),
    totalDurationNs: core.serialization.property("total_duration_ns", core.serialization.number().optional()),
});
