// @letta-ai/letta-client/serialization/types/StepMetrics.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.StepMetrics = void 0;
const core = __importStar(require("../../core"));
exports.StepMetrics = core.serialization.object({
    id: core.serialization.string(),
    providerId: core.serialization.property("provider_id", core.serialization.string().optional()),
    runId: core.serialization.property("run_id", core.serialization.string().optional()),
    agentId: core.serialization.property("agent_id", core.serialization.string().optional()),
    stepStartNs: core.serialization.property("step_start_ns", core.serialization.number().optional()),
    llmRequestStartNs: core.serialization.property("llm_request_start_ns", core.serialization.number().optional()),
    llmRequestNs: core.serialization.property("llm_request_ns", core.serialization.number().optional()),
    toolExecutionNs: core.serialization.property("tool_execution_ns", core.serialization.number().optional()),
    stepNs: core.serialization.property("step_ns", core.serialization.number().optional()),
    baseTemplateId: core.serialization.property("base_template_id", core.serialization.string().optional()),
    templateId: core.serialization.property("template_id", core.serialization.string().optional()),
    projectId: core.serialization.property("project_id", core.serialization.string().optional()),
});
