// @letta-ai/letta-client/serialization/resources/tools/client/requests/ToolRunFromSource.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ToolRunFromSource = void 0;
const core = __importStar(require("../../../../../core"));
const PipRequirement_1 = require("../../../../types/PipRequirement");
const NpmRequirement_1 = require("../../../../types/NpmRequirement");
exports.ToolRunFromSource = core.serialization.object({
    sourceCode: core.serialization.property("source_code", core.serialization.string()),
    args: core.serialization.record(core.serialization.string(), core.serialization.unknown()),
    envVars: core.serialization.property("env_vars", core.serialization.record(core.serialization.string(), core.serialization.string()).optional()),
    name: core.serialization.string().optional(),
    sourceType: core.serialization.property("source_type", core.serialization.string().optional()),
    argsJsonSchema: core.serialization.property("args_json_schema", core.serialization.record(core.serialization.string(), core.serialization.unknown()).optional()),
    jsonSchema: core.serialization.property("json_schema", core.serialization.record(core.serialization.string(), core.serialization.unknown()).optional()),
    pipRequirements: core.serialization.property("pip_requirements", core.serialization.list(PipRequirement_1.PipRequirement).optional()),
    npmRequirements: core.serialization.property("npm_requirements", core.serialization.list(NpmRequirement_1.NpmRequirement).optional()),
});
