// @letta-ai/letta-client/dist/serialization/types/AgentState.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AgentState = void 0;
const core = __importStar(require("../../core"));
const AgentStateToolRulesItem_1 = require("./AgentStateToolRulesItem");
const AgentType_1 = require("./AgentType");
const LlmConfig_1 = require("./LlmConfig");
const EmbeddingConfig_1 = require("./EmbeddingConfig");
const AgentStateResponseFormat_1 = require("./AgentStateResponseFormat");
const Memory_1 = require("./Memory");
const Tool_1 = require("./Tool");
const Source_1 = require("./Source");
const AgentEnvironmentVariable_1 = require("./AgentEnvironmentVariable");
const Group_1 = require("./Group");
exports.AgentState = core.serialization.object({
    createdById: core.serialization.property("created_by_id", core.serialization.string().optional()),
    lastUpdatedById: core.serialization.property("last_updated_by_id", core.serialization.string().optional()),
    createdAt: core.serialization.property("created_at", core.serialization.date().optional()),
    updatedAt: core.serialization.property("updated_at", core.serialization.date().optional()),
    id: core.serialization.string(),
    name: core.serialization.string(),
    toolRules: core.serialization.property("tool_rules", core.serialization.list(AgentStateToolRulesItem_1.AgentStateToolRulesItem).optional()),
    messageIds: core.serialization.property("message_ids", core.serialization.list(core.serialization.string()).optional()),
    system: core.serialization.string(),
    agentType: core.serialization.property("agent_type", AgentType_1.AgentType),
    llmConfig: core.serialization.property("llm_config", LlmConfig_1.LlmConfig),
    embeddingConfig: core.serialization.property("embedding_config", EmbeddingConfig_1.EmbeddingConfig),
    responseFormat: core.serialization.property("response_format", AgentStateResponseFormat_1.AgentStateResponseFormat.optional()),
    description: core.serialization.string().optional(),
    metadata: core.serialization.record(core.serialization.string(), core.serialization.unknown()).optional(),
    memory: Memory_1.Memory,
    tools: core.serialization.list(Tool_1.Tool),
    sources: core.serialization.list(Source_1.Source),
    tags: core.serialization.list(core.serialization.string()),
    toolExecEnvironmentVariables: core.serialization.property("tool_exec_environment_variables", core.serialization.list(AgentEnvironmentVariable_1.AgentEnvironmentVariable).optional()),
    secrets: core.serialization.list(AgentEnvironmentVariable_1.AgentEnvironmentVariable).optional(),
    projectId: core.serialization.property("project_id", core.serialization.string().optional()),
    templateId: core.serialization.property("template_id", core.serialization.string().optional()),
    baseTemplateId: core.serialization.property("base_template_id", core.serialization.string().optional()),
    deploymentId: core.serialization.property("deployment_id", core.serialization.string().optional()),
    entityId: core.serialization.property("entity_id", core.serialization.string().optional()),
    identityIds: core.serialization.property("identity_ids", core.serialization.list(core.serialization.string()).optional()),
    messageBufferAutoclear: core.serialization.property("message_buffer_autoclear", core.serialization.boolean().optional()),
    enableSleeptime: core.serialization.property("enable_sleeptime", core.serialization.boolean().optional()),
    multiAgentGroup: core.serialization.property("multi_agent_group", Group_1.Group.optional()),
    lastRunCompletion: core.serialization.property("last_run_completion", core.serialization.date().optional()),
    lastRunDurationMs: core.serialization.property("last_run_duration_ms", core.serialization.number().optional()),
    timezone: core.serialization.string().optional(),
    maxFilesOpen: core.serialization.property("max_files_open", core.serialization.number().optional()),
    perFileViewWindowCharLimit: core.serialization.property("per_file_view_window_char_limit", core.serialization.number().optional()),
    hidden: core.serialization.boolean().optional(),
});
