// @letta-ai/letta-client/dist/serialization/types/FileSchema.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.FileSchema = void 0;
const core = __importStar(require("../../core"));
const FileProcessingStatus_1 = require("./FileProcessingStatus");
exports.FileSchema = core.serialization.object({
    sourceId: core.serialization.property("source_id", core.serialization.string()),
    fileName: core.serialization.property("file_name", core.serialization.string().optional()),
    originalFileName: core.serialization.property("original_file_name", core.serialization.string().optional()),
    filePath: core.serialization.property("file_path", core.serialization.string().optional()),
    fileType: core.serialization.property("file_type", core.serialization.string().optional()),
    fileSize: core.serialization.property("file_size", core.serialization.number().optional()),
    fileCreationDate: core.serialization.property("file_creation_date", core.serialization.string().optional()),
    fileLastModifiedDate: core.serialization.property("file_last_modified_date", core.serialization.string().optional()),
    processingStatus: core.serialization.property("processing_status", FileProcessingStatus_1.FileProcessingStatus.optional()),
    errorMessage: core.serialization.property("error_message", core.serialization.string().optional()),
    totalChunks: core.serialization.property("total_chunks", core.serialization.number().optional()),
    chunksEmbedded: core.serialization.property("chunks_embedded", core.serialization.number().optional()),
    content: core.serialization.string().optional(),
    id: core.serialization.string(),
});
