// @letta-ai/letta-client/dist/serialization/types/Group.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Group = void 0;
const core = __importStar(require("../../core"));
const ManagerType_1 = require("./ManagerType");
exports.Group = core.serialization.object({
    id: core.serialization.string(),
    managerType: core.serialization.property("manager_type", ManagerType_1.ManagerType),
    agentIds: core.serialization.property("agent_ids", core.serialization.list(core.serialization.string())),
    description: core.serialization.string(),
    projectId: core.serialization.property("project_id", core.serialization.string().optional()),
    templateId: core.serialization.property("template_id", core.serialization.string().optional()),
    baseTemplateId: core.serialization.property("base_template_id", core.serialization.string().optional()),
    deploymentId: core.serialization.property("deployment_id", core.serialization.string().optional()),
    sharedBlockIds: core.serialization.property("shared_block_ids", core.serialization.list(core.serialization.string()).optional()),
    managerAgentId: core.serialization.property("manager_agent_id", core.serialization.string().optional()),
    terminationToken: core.serialization.property("termination_token", core.serialization.string().optional()),
    maxTurns: core.serialization.property("max_turns", core.serialization.number().optional()),
    sleeptimeAgentFrequency: core.serialization.property("sleeptime_agent_frequency", core.serialization.number().optional()),
    turnsCounter: core.serialization.property("turns_counter", core.serialization.number().optional()),
    lastProcessedMessageId: core.serialization.property("last_processed_message_id", core.serialization.string().optional()),
    maxMessageBufferLength: core.serialization.property("max_message_buffer_length", core.serialization.number().optional()),
    minMessageBufferLength: core.serialization.property("min_message_buffer_length", core.serialization.number().optional()),
    hidden: core.serialization.boolean().optional(),
});
