// @letta-ai/letta-client/dist/serialization/types/LettaSchemasAgentFileMessageSchema.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LettaSchemasAgentFileMessageSchema = void 0;
const core = __importStar(require("../../core"));
const MessageRole_1 = require("./MessageRole");
const LettaSchemasAgentFileMessageSchemaContent_1 = require("./LettaSchemasAgentFileMessageSchemaContent");
const ChatCompletionMessageFunctionToolCallInput_1 = require("./ChatCompletionMessageFunctionToolCallInput");
const LettaSchemasMessageToolReturn_1 = require("./LettaSchemasMessageToolReturn");
const LettaSchemasAgentFileMessageSchemaApprovalsItem_1 = require("./LettaSchemasAgentFileMessageSchemaApprovalsItem");
exports.LettaSchemasAgentFileMessageSchema = core.serialization.object({
    type: core.serialization.stringLiteral("message").optional(),
    role: MessageRole_1.MessageRole,
    content: LettaSchemasAgentFileMessageSchemaContent_1.LettaSchemasAgentFileMessageSchemaContent,
    name: core.serialization.string().optional(),
    otid: core.serialization.string().optional(),
    senderId: core.serialization.property("sender_id", core.serialization.string().optional()),
    batchItemId: core.serialization.property("batch_item_id", core.serialization.string().optional()),
    groupId: core.serialization.property("group_id", core.serialization.string().optional()),
    id: core.serialization.string(),
    model: core.serialization.string().optional(),
    agentId: core.serialization.property("agent_id", core.serialization.string().optional()),
    toolCalls: core.serialization.property("tool_calls", core.serialization.list(ChatCompletionMessageFunctionToolCallInput_1.ChatCompletionMessageFunctionToolCallInput).optional()),
    toolCallId: core.serialization.property("tool_call_id", core.serialization.string().optional()),
    toolReturns: core.serialization.property("tool_returns", core.serialization.list(LettaSchemasMessageToolReturn_1.LettaSchemasMessageToolReturn).optional()),
    createdAt: core.serialization.property("created_at", core.serialization.date().optional()),
    approve: core.serialization.boolean().optional(),
    approvalRequestId: core.serialization.property("approval_request_id", core.serialization.string().optional()),
    denialReason: core.serialization.property("denial_reason", core.serialization.string().optional()),
    approvals: core.serialization.list(LettaSchemasAgentFileMessageSchemaApprovalsItem_1.LettaSchemasAgentFileMessageSchemaApprovalsItem).optional(),
});
