// @letta-ai/letta-client/dist/serialization/types/LlmConfig.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.LlmConfig = void 0;
const core = __importStar(require("../../core"));
const LlmConfigModelEndpointType_1 = require("./LlmConfigModelEndpointType");
const ProviderCategory_1 = require("./ProviderCategory");
const LlmConfigReasoningEffort_1 = require("./LlmConfigReasoningEffort");
const LlmConfigCompatibilityType_1 = require("./LlmConfigCompatibilityType");
const LlmConfigVerbosity_1 = require("./LlmConfigVerbosity");
exports.LlmConfig = core.serialization.object({
    model: core.serialization.string(),
    displayName: core.serialization.property("display_name", core.serialization.string().optional()),
    modelEndpointType: core.serialization.property("model_endpoint_type", LlmConfigModelEndpointType_1.LlmConfigModelEndpointType),
    modelEndpoint: core.serialization.property("model_endpoint", core.serialization.string().optional()),
    providerName: core.serialization.property("provider_name", core.serialization.string().optional()),
    providerCategory: core.serialization.property("provider_category", ProviderCategory_1.ProviderCategory.optional()),
    modelWrapper: core.serialization.property("model_wrapper", core.serialization.string().optional()),
    contextWindow: core.serialization.property("context_window", core.serialization.number()),
    putInnerThoughtsInKwargs: core.serialization.property("put_inner_thoughts_in_kwargs", core.serialization.boolean().optional()),
    handle: core.serialization.string().optional(),
    temperature: core.serialization.number().optional(),
    maxTokens: core.serialization.property("max_tokens", core.serialization.number().optional()),
    enableReasoner: core.serialization.property("enable_reasoner", core.serialization.boolean().optional()),
    reasoningEffort: core.serialization.property("reasoning_effort", LlmConfigReasoningEffort_1.LlmConfigReasoningEffort.optional()),
    maxReasoningTokens: core.serialization.property("max_reasoning_tokens", core.serialization.number().optional()),
    frequencyPenalty: core.serialization.property("frequency_penalty", core.serialization.number().optional()),
    compatibilityType: core.serialization.property("compatibility_type", LlmConfigCompatibilityType_1.LlmConfigCompatibilityType.optional()),
    verbosity: LlmConfigVerbosity_1.LlmConfigVerbosity.optional(),
    tier: core.serialization.string().optional(),
    parallelToolCalls: core.serialization.property("parallel_tool_calls", core.serialization.boolean().optional()),
});
