// @letta-ai/letta-client/dist/serialization/types/Step.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Step = void 0;
const core = __importStar(require("../../core"));
const StopReasonType_1 = require("./StopReasonType");
const Message_1 = require("./Message");
const StepFeedback_1 = require("./StepFeedback");
const StepStatus_1 = require("./StepStatus");
exports.Step = core.serialization.object({
    id: core.serialization.string(),
    origin: core.serialization.string().optional(),
    providerId: core.serialization.property("provider_id", core.serialization.string().optional()),
    runId: core.serialization.property("run_id", core.serialization.string().optional()),
    agentId: core.serialization.property("agent_id", core.serialization.string().optional()),
    providerName: core.serialization.property("provider_name", core.serialization.string().optional()),
    providerCategory: core.serialization.property("provider_category", core.serialization.string().optional()),
    model: core.serialization.string().optional(),
    modelEndpoint: core.serialization.property("model_endpoint", core.serialization.string().optional()),
    contextWindowLimit: core.serialization.property("context_window_limit", core.serialization.number().optional()),
    completionTokens: core.serialization.property("completion_tokens", core.serialization.number().optional()),
    promptTokens: core.serialization.property("prompt_tokens", core.serialization.number().optional()),
    totalTokens: core.serialization.property("total_tokens", core.serialization.number().optional()),
    completionTokensDetails: core.serialization.property("completion_tokens_details", core.serialization.record(core.serialization.string(), core.serialization.unknown()).optional()),
    stopReason: core.serialization.property("stop_reason", StopReasonType_1.StopReasonType.optional()),
    tags: core.serialization.list(core.serialization.string()).optional(),
    tid: core.serialization.string().optional(),
    traceId: core.serialization.property("trace_id", core.serialization.string().optional()),
    messages: core.serialization.list(Message_1.Message).optional(),
    feedback: StepFeedback_1.StepFeedback.optional(),
    projectId: core.serialization.property("project_id", core.serialization.string().optional()),
    errorType: core.serialization.property("error_type", core.serialization.string().optional()),
    errorData: core.serialization.property("error_data", core.serialization.record(core.serialization.string(), core.serialization.unknown()).optional()),
    status: StepStatus_1.StepStatus.optional(),
});
