// @letta-ai/letta-client/api/resources/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.voice = exports.telemetry = exports.health = exports.models = exports.sources = exports.projects = exports.clientSideAccessTokens = exports.templates = exports.batches = exports.tags = exports.steps = exports.runs = exports.providers = exports.jobs = exports.blocks = exports.identities = exports.groups = exports.chat = exports.agents = exports.folders = exports.tools = exports.archives = void 0;
exports.archives = __importStar(require("./archives"));
__exportStar(require("./archives/types"), exports);
exports.tools = __importStar(require("./tools"));
__exportStar(require("./tools/types"), exports);
exports.folders = __importStar(require("./folders"));
__exportStar(require("./folders/types"), exports);
exports.agents = __importStar(require("./agents"));
__exportStar(require("./agents/types"), exports);
exports.chat = __importStar(require("./chat"));
__exportStar(require("./chat/types"), exports);
exports.groups = __importStar(require("./groups"));
__exportStar(require("./groups/types"), exports);
exports.identities = __importStar(require("./identities"));
__exportStar(require("./identities/types"), exports);
exports.blocks = __importStar(require("./blocks"));
__exportStar(require("./blocks/types"), exports);
exports.jobs = __importStar(require("./jobs"));
__exportStar(require("./jobs/types"), exports);
exports.providers = __importStar(require("./providers"));
__exportStar(require("./providers/types"), exports);
exports.runs = __importStar(require("./runs"));
__exportStar(require("./runs/types"), exports);
exports.steps = __importStar(require("./steps"));
__exportStar(require("./steps/types"), exports);
exports.tags = __importStar(require("./tags"));
__exportStar(require("./tags/types"), exports);
exports.batches = __importStar(require("./batches"));
__exportStar(require("./batches/types"), exports);
exports.templates = __importStar(require("./templates"));
__exportStar(require("./templates/types"), exports);
exports.clientSideAccessTokens = __importStar(require("./clientSideAccessTokens"));
__exportStar(require("./clientSideAccessTokens/types"), exports);
exports.projects = __importStar(require("./projects"));
__exportStar(require("./projects/types"), exports);
exports.sources = __importStar(require("./sources"));
exports.models = __importStar(require("./models"));
exports.health = __importStar(require("./health"));
exports.telemetry = __importStar(require("./telemetry"));
exports.voice = __importStar(require("./voice"));
__exportStar(require("./archives/client/requests"), exports);
__exportStar(require("./tools/client/requests"), exports);
__exportStar(require("./sources/client/requests"), exports);
__exportStar(require("./folders/client/requests"), exports);
__exportStar(require("./agents/client/requests"), exports);
__exportStar(require("./chat/client/requests"), exports);
__exportStar(require("./groups/client/requests"), exports);
__exportStar(require("./identities/client/requests"), exports);
__exportStar(require("./models/client/requests"), exports);
__exportStar(require("./blocks/client/requests"), exports);
__exportStar(require("./jobs/client/requests"), exports);
__exportStar(require("./providers/client/requests"), exports);
__exportStar(require("./runs/client/requests"), exports);
__exportStar(require("./steps/client/requests"), exports);
__exportStar(require("./tags/client/requests"), exports);
__exportStar(require("./batches/client/requests"), exports);
__exportStar(require("./templates/client/requests"), exports);
__exportStar(require("./clientSideAccessTokens/client/requests"), exports);
__exportStar(require("./projects/client/requests"), exports);
