// @letta-ai/letta-client/dist/serialization/resources/templates/types/TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfig.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfig = void 0;
const core = __importStar(require("../../../../core"));
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingEndpointType_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingEndpointType");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingEndpoint_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingEndpoint");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingChunkSize_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingChunkSize");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigHandle_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigHandle");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigBatchSize_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigBatchSize");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigAzureEndpoint_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigAzureEndpoint");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigAzureVersion_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigAzureVersion");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigAzureDeployment_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigAzureDeployment");
exports.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfig = core.serialization.object({
    embeddingEndpointType: core.serialization.property("embedding_endpoint_type", TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingEndpointType_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingEndpointType),
    embeddingEndpoint: core.serialization.property("embedding_endpoint", TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingEndpoint_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingEndpoint.optional()),
    embeddingModel: core.serialization.property("embedding_model", core.serialization.string()),
    embeddingDim: core.serialization.property("embedding_dim", core.serialization.number()),
    embeddingChunkSize: core.serialization.property("embedding_chunk_size", TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingChunkSize_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigEmbeddingChunkSize.optional()),
    handle: TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigHandle_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigHandle.optional(),
    batchSize: core.serialization.property("batch_size", TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigBatchSize_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigBatchSize.optional()),
    azureEndpoint: core.serialization.property("azure_endpoint", TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigAzureEndpoint_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigAzureEndpoint.optional()),
    azureVersion: core.serialization.property("azure_version", TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigAzureVersion_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigAzureVersion.optional()),
    azureDeployment: core.serialization.property("azure_deployment", TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigAzureDeployment_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSourcesItemEmbeddingConfigAzureDeployment.optional()),
});
