// @letta-ai/letta-client/dist/serialization/resources/templates/types/TemplatesGetTemplateSnapshotResponseAgentsItemProperties.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplatesGetTemplateSnapshotResponseAgentsItemProperties = void 0;
const core = __importStar(require("../../../../core"));
const TemplatesGetTemplateSnapshotResponseAgentsItemPropertiesVerbosityLevel_1 = require("./TemplatesGetTemplateSnapshotResponseAgentsItemPropertiesVerbosityLevel");
const TemplatesGetTemplateSnapshotResponseAgentsItemPropertiesReasoningEffort_1 = require("./TemplatesGetTemplateSnapshotResponseAgentsItemPropertiesReasoningEffort");
exports.TemplatesGetTemplateSnapshotResponseAgentsItemProperties = core.serialization.object({
    enableReasoner: core.serialization.property("enable_reasoner", core.serialization.boolean().optional()),
    putInnerThoughtsInKwargs: core.serialization.property("put_inner_thoughts_in_kwargs", core.serialization.boolean().optional()),
    contextWindowLimit: core.serialization.property("context_window_limit", core.serialization.number().optional()),
    maxTokens: core.serialization.property("max_tokens", core.serialization.number().optional()),
    maxReasoningTokens: core.serialization.property("max_reasoning_tokens", core.serialization.number().optional()),
    maxFilesOpen: core.serialization.property("max_files_open", core.serialization.number().optional()),
    messageBufferAutoclear: core.serialization.property("message_buffer_autoclear", core.serialization.boolean().optional()),
    verbosityLevel: core.serialization.property("verbosity_level", TemplatesGetTemplateSnapshotResponseAgentsItemPropertiesVerbosityLevel_1.TemplatesGetTemplateSnapshotResponseAgentsItemPropertiesVerbosityLevel.optional()),
    reasoningEffort: core.serialization.property("reasoning_effort", TemplatesGetTemplateSnapshotResponseAgentsItemPropertiesReasoningEffort_1.TemplatesGetTemplateSnapshotResponseAgentsItemPropertiesReasoningEffort.optional()),
    perFileViewWindowCharLimit: core.serialization.property("per_file_view_window_char_limit", core.serialization.number().optional()),
    temperature: core.serialization.number().optional(),
});
