// @letta-ai/letta-client/dist/serialization/resources/templates/types/TemplatesGetTemplateSnapshotResponse.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplatesGetTemplateSnapshotResponse = void 0;
const core = __importStar(require("../../../../core"));
const TemplatesGetTemplateSnapshotResponseAgentsItem_1 = require("./TemplatesGetTemplateSnapshotResponseAgentsItem");
const TemplatesGetTemplateSnapshotResponseBlocksItem_1 = require("./TemplatesGetTemplateSnapshotResponseBlocksItem");
const TemplatesGetTemplateSnapshotResponseRelationshipsItem_1 = require("./TemplatesGetTemplateSnapshotResponseRelationshipsItem");
const TemplatesGetTemplateSnapshotResponseConfiguration_1 = require("./TemplatesGetTemplateSnapshotResponseConfiguration");
const TemplatesGetTemplateSnapshotResponseType_1 = require("./TemplatesGetTemplateSnapshotResponseType");
exports.TemplatesGetTemplateSnapshotResponse = core.serialization.object({
    agents: core.serialization.list(TemplatesGetTemplateSnapshotResponseAgentsItem_1.TemplatesGetTemplateSnapshotResponseAgentsItem),
    blocks: core.serialization.list(TemplatesGetTemplateSnapshotResponseBlocksItem_1.TemplatesGetTemplateSnapshotResponseBlocksItem),
    relationships: core.serialization.list(TemplatesGetTemplateSnapshotResponseRelationshipsItem_1.TemplatesGetTemplateSnapshotResponseRelationshipsItem),
    configuration: TemplatesGetTemplateSnapshotResponseConfiguration_1.TemplatesGetTemplateSnapshotResponseConfiguration,
    type: TemplatesGetTemplateSnapshotResponseType_1.TemplatesGetTemplateSnapshotResponseType,
    version: core.serialization.string(),
});
