// @letta-ai/letta-client/dist/serialization/resources/templates/types/TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfig.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfig = void 0;
const core = __importStar(require("../../../../core"));
const TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigEmbeddingEndpointType_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigEmbeddingEndpointType");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigEmbeddingEndpoint_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigEmbeddingEndpoint");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigEmbeddingChunkSize_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigEmbeddingChunkSize");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigHandle_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigHandle");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigBatchSize_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigBatchSize");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigAzureEndpoint_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigAzureEndpoint");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigAzureVersion_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigAzureVersion");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigAzureDeployment_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigAzureDeployment");
exports.TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfig = core.serialization.object({
    embeddingEndpointType: core.serialization.property("embedding_endpoint_type", TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigEmbeddingEndpointType_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigEmbeddingEndpointType),
    embeddingEndpoint: core.serialization.property("embedding_endpoint", TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigEmbeddingEndpoint_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigEmbeddingEndpoint.optional()),
    embeddingModel: core.serialization.property("embedding_model", core.serialization.string()),
    embeddingDim: core.serialization.property("embedding_dim", core.serialization.number()),
    embeddingChunkSize: core.serialization.property("embedding_chunk_size", TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigEmbeddingChunkSize_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigEmbeddingChunkSize.optional()),
    handle: TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigHandle_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigHandle.optional(),
    batchSize: core.serialization.property("batch_size", TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigBatchSize_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigBatchSize.optional()),
    azureEndpoint: core.serialization.property("azure_endpoint", TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigAzureEndpoint_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigAzureEndpoint.optional()),
    azureVersion: core.serialization.property("azure_version", TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigAzureVersion_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigAzureVersion.optional()),
    azureDeployment: core.serialization.property("azure_deployment", TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigAzureDeployment_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemEmbeddingConfigAzureDeployment.optional()),
});
