// @letta-ai/letta-client/dist/serialization/resources/templates/types/TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItem.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItem = void 0;
const core = __importStar(require("../../../../core"));
const TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemCreatedById_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemCreatedById");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemLastUpdatedById_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemLastUpdatedById");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemCreatedAt_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemCreatedAt");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemUpdatedAt_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemUpdatedAt");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemId_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemId");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemDescription_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemDescription");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemOrganizationId_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemOrganizationId");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemValueEnc_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemValueEnc");
exports.TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItem = core.serialization.object({
    createdById: core.serialization.property("created_by_id", TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemCreatedById_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemCreatedById.optional()),
    lastUpdatedById: core.serialization.property("last_updated_by_id", TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemLastUpdatedById_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemLastUpdatedById.optional()),
    createdAt: core.serialization.property("created_at", TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemCreatedAt_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemCreatedAt.optional()),
    updatedAt: core.serialization.property("updated_at", TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemUpdatedAt_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemUpdatedAt.optional()),
    id: TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemId_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemId.optional(),
    key: core.serialization.string(),
    value: core.serialization.string(),
    description: TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemDescription_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemDescription.optional(),
    organizationId: core.serialization.property("organization_id", TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemOrganizationId_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemOrganizationId.optional()),
    valueEnc: core.serialization.property("value_enc", TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemValueEnc_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemToolExecEnvironmentVariablesItemValueEnc.optional()),
    agentId: core.serialization.property("agent_id", core.serialization.string()),
});
