// @letta-ai/letta-client/dist/serialization/resources/templates/types/TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMapping.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMapping = void 0;
const core = __importStar(require("../../../../core"));
const TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMappingType_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMappingType");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMappingPromptTemplate_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMappingPromptTemplate");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMappingDefaultChild_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMappingDefaultChild");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMappingRequireOutputMapping_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMappingRequireOutputMapping");
exports.TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMapping = core.serialization.object({
    toolName: core.serialization.property("tool_name", core.serialization.string()),
    type: TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMappingType_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMappingType.optional(),
    promptTemplate: core.serialization.property("prompt_template", TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMappingPromptTemplate_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMappingPromptTemplate.optional()),
    defaultChild: core.serialization.property("default_child", TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMappingDefaultChild_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMappingDefaultChild.optional()),
    childOutputMapping: core.serialization.property("child_output_mapping", core.serialization.unknown().optional()),
    requireOutputMapping: core.serialization.property("require_output_mapping", TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMappingRequireOutputMapping_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemToolRulesItemChildOutputMappingRequireOutputMapping.optional()),
});
