// @letta-ai/letta-client/dist/serialization/resources/templates/types/TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItem.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItem = void 0;
const core = __importStar(require("../../../../core"));
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemCreatedById_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemCreatedById");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemLastUpdatedById_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemLastUpdatedById");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemCreatedAt_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemCreatedAt");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemUpdatedAt_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemUpdatedAt");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemId_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemId");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemDescription_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemDescription");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemOrganizationId_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemOrganizationId");
const TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemValueEnc_1 = require("./TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemValueEnc");
exports.TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItem = core.serialization.object({
    createdById: core.serialization.property("created_by_id", TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemCreatedById_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemCreatedById.optional()),
    lastUpdatedById: core.serialization.property("last_updated_by_id", TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemLastUpdatedById_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemLastUpdatedById.optional()),
    createdAt: core.serialization.property("created_at", TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemCreatedAt_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemCreatedAt.optional()),
    updatedAt: core.serialization.property("updated_at", TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemUpdatedAt_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemUpdatedAt.optional()),
    id: TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemId_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemId.optional(),
    key: core.serialization.string(),
    value: core.serialization.string(),
    description: TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemDescription_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemDescription.optional(),
    organizationId: core.serialization.property("organization_id", TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemOrganizationId_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemOrganizationId.optional()),
    valueEnc: core.serialization.property("value_enc", TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemValueEnc_1.TemplatesCreateAgentsFromTemplateResponseAgentsItemSecretsItemValueEnc.optional()),
    agentId: core.serialization.property("agent_id", core.serialization.string()),
});
